cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -1222,23 +1365,21 @@ impl ABICall for AArch64ABICall {
     type I = Inst;
 
     fn num_args(&self) -> usize {
-        self.sig.args.len()
+        if self.sig.stack_ret_arg.is_some() {
+            self.sig.args.len() - 1
+        } else {
+            self.sig.args.len()
+        }
     }
 
     fn emit_stack_pre_adjust<C: LowerCtx<I = Self::I>>(&self, ctx: &mut C) {
-        adjust_stack(
-            ctx,
-            self.sig.stack_arg_space as u64,
-            /* is_sub = */ true,
-        )
+        let off = self.sig.stack_arg_space + self.sig.stack_ret_space;
```

Comment:
Do we want to check for overflow here?

Reply:
The u32 + u32 you mean? I think that wasm implementation limits will prevent the arg space + ret space from reaching 2GB (or getting close), but maybe I'm misunderstanding which overflow case you mean?

Reply:
WASM may prevent it, but other users may not impose such constraints themselves.

Reply:
Yes, what @bjorn3 said (thanks!).

Reply:
Done -- added a reasonable limit (128MB) to the `ABISig` construction so this is guaranteed not to overflow. Thanks!